src/lib/updating-element.ts

```diff
@@ -0,0 +1,552 @@
+   * invalidation and update.
+   */
+  static createProperty(name: PropertyKey, options: PropertyDeclaration = defaultPropertyDeclaration) {
+    // ensure private storage for property declarations.
+    if (!this.hasOwnProperty('_classProperties')) {
+      this._classProperties = new Map();
+      // NOTE: Workaround IE11 not supporting Map constructor argument.
+      const superProperties = Object.getPrototypeOf(this)._classProperties;
+      if (superProperties !== undefined) {
+        superProperties.forEach((v: any, k: PropertyKey) =>
+          this._classProperties.set(k, v));
+      }
+    }
+    this._classProperties.set(name, options);
+    // Allow user defined accessors by not replacing an existing own-property accessor.
+    if (this.prototype.hasOwnProperty(name)) {
+      return;
+    }
+    const key = typeof name === 'symbol' ? Symbol() : `__${name}`;
+    Object.defineProperty(this.prototype, name, {
+      get() {
+        return this[key];
+      },
+      set(value) {
+        const oldValue = this[name];
+        this[key] = value;
+        this.invalidateProperty(name, oldValue, options);
+      },
+      configurable: true,
+      enumerable: true
+    });
+  }
+
+  /**
+   * Creates property accessors for registered properties and ensures
+   * any superclasses are also finalized.
+   */
+  private static _finalize() {
+    if (this.hasOwnProperty('_finalized') && this._finalized) {
+      return;
+    }
+    // finalize any superclasses
+    const superCtor = Object.getPrototypeOf(this);
+    if (typeof superCtor._finalize === 'function') {
+      superCtor._finalize();
+    }
+    this._finalized = true;
+    // initialize map populated in observedAttributes
+    this._attributeToPropertyMap = new Map();
+    // make any properties
+    const props = this.properties;
+    // support symbols in properties (IE11 does not support this)
+    const propKeys = [...Object.getOwnPropertyNames(props),
+        ...(typeof Object.getOwnPropertySymbols === 'function') ? Object.getOwnPropertySymbols(props) : []];
+    for (const p of propKeys) {
+      // note, use of `any` is due to TypeSript lack of support for symbol in index types
+      this.createProperty(p, (props as any)[p]);
+    }
+  }
+
+  /**
+   * Returns the property name for the given attribute `name`.
+   */
+  private static _attributeNameForProperty(name: PropertyKey, options?: PropertyDeclaration) {
+    const attribute = options !== undefined && options.attribute;
+    return attribute === false ? undefined : (typeof attribute === 'string' ?
+        attribute : (typeof name === 'string' ? name.toLowerCase() : undefined));
+  }
+
+  /**
+   * Returns true if a property should cause invalidation and update.
+   * Called when a property value is set and uses the `shouldInvalidate`
+   * option for the property if present or a strict identity check.
+   */
+  private static _propertyShouldInvalidate(value: unknown, old: unknown,
```

Comment:
Since we don't have the property name here, I'd consider a name like `_valueShouldInvalidate`, as `_propertyShouldInvalidate` sounds like the function you'd get for a specific property (the third arg here).

Reply:
Done.